File: src/main/java/play/pluv/security/AuthorizationExtractor.java
package play.pluv.security;

import static org.springframework.http.HttpHeaders.AUTHORIZATION;
import static play.pluv.security.exception.SecurityExceptionType.INVALID_ACCESS_TOKEN_TYPE;
import static play.pluv.security.exception.SecurityExceptionType.NOT_FOUND_AUTHORIZATION_TOKEN;

import jakarta.servlet.http.HttpServletRequest;
import play.pluv.login.exception.LoginException;
import play.pluv.security.exception.SecurityException;

public class AuthorizationExtractor {

  private static final String BEARER_TYPE = "Bearer";

  public static String extract(final HttpServletRequest request) {
    final String authorizationHeader = request.getHeader(AUTHORIZATION);

    validateAuthorizationHeader(authorizationHeader);

    return authorizationHeader.substring(BEARER_TYPE.length()).trim();
  }

  private static void validateAuthorizationHeader(final String authorizationHeader) {
    if (authorizationHeader == null || authorizationHeader.isBlank()) {
      throw new SecurityException(NOT_FOUND_AUTHORIZATION_TOKEN);
    }
    if (!authorizationHeader.toLowerCase().startsWith(BEARER_TYPE.toLowerCase())) {
      throw new SecurityException(INVALID_ACCESS_TOKEN_TYPE);
    }
  }
}

File: src/main/kotlin/play/pluv/security/AuthorizationExtractor.kt
package play.pluv.security

import jakarta.servlet.http.HttpServletRequest
import org.springframework.http.HttpHeaders.AUTHORIZATION
import play.pluv.security.exception.SecurityException
import play.pluv.security.exception.SecurityExceptionType.INVALID_ACCESS_TOKEN_TYPE
import play.pluv.security.exception.SecurityExceptionType.NOT_FOUND_AUTHORIZATION_TOKEN

class AuthorizationExtractor {

    companion object {
        private const val BEARER_TYPE = "Bearer"

        fun extract(request: HttpServletRequest): String {
            val authorizationHeader = request.getHeader(AUTHORIZATION)

            validateAuthorizationHeader(authorizationHeader)

            return authorizationHeader.substring(BEARER_TYPE.length).trim()
        }

        private fun validateAuthorizationHeader(authorizationHeader: String) {
            if (authorizationHeader.isBlank()) {
                throw SecurityException(NOT_FOUND_AUTHORIZATION_TOKEN)
            }
            if (!authorizationHeader.startsWith(BEARER_TYPE)) {
                throw SecurityException(INVALID_ACCESS_TOKEN_TYPE)
            }
        }
    }
}